main.go

```diff
@@ -71,79 +71,112 @@ func failf(format string, v ...interface{}) {
 	os.Exit(1)
 }
 
-func main() {
-	var cfg config
-	if err := stepconf.Parse(&cfg); err != nil {
-		failf("Issue with input: %s", err)
-	}
-	stepconf.Print(cfg)
-	fmt.Println()
-
-	if cfg.NewVersionName == nil && cfg.NewVersionCode == nil {
-		failf("Neither NewVersionCode nor NewVersionName are provided, however one of them is required.")
-	}
+// BuildGradleVersionUpdater updates versionName and versionCode in the given build.gradle file.
+type BuildGradleVersionUpdater struct {
+	buildGradleReader io.Reader
+}
 
-	//
-	// find versionName & versionCode with regexp
-	fmt.Println()
-	log.Infof("Updating versionName and versionCode in: %s", cfg.BuildGradlePth)
+// NewBuildGradleVersionUpdater constructs a new BuildGradleVersionUpdater.
+func NewBuildGradleVersionUpdater(buildGradleReader io.Reader) BuildGradleVersionUpdater {
+	return BuildGradleVersionUpdater{buildGradleReader: buildGradleReader}
+}
 
-	f, err := os.Open(cfg.BuildGradlePth)
-	if err != nil {
-		failf("Failed to read build.gradle file, error: %s", err)
-	}
+// UpdateResult stors the result of the version update.
+type UpdateResult struct {
+	NewContent                               string
+	FinalVersionCode, FinalVersionName       string
+	UpdatedVersionCodes, UpdatedVersionNames int
+}
 
-	var finalVersionCode, finalVersionName string
-	var updatedVersionCodes, updatedVersionNames int
+// UpdateVersion executes the version updates.
+func (u BuildGradleVersionUpdater) UpdateVersion(newVersionCode *int, versionCodeOffset int, newVersionName *string) (*UpdateResult, error) {
```

Comment:
Why pointers for `newVersionCode` and `newVersionName` my spider sense triggers.

Also why the pointer return value?

Reply:
Pointer arguments: initially I just wanted to move the update functionality into a new function, without modifying it. Updated the code to use value arguments.

Pointer return value: my intention was to use the `nil` return value if the update fails. I dropped this idea, as the function not just updating, but reading and exposing the versionCode and versionName (even if nothing was updated).